tapestry.php

```diff
@@ -354,3 +354,93 @@ function prefix_title_entity_decode($response)
 
     return $response;
 }
+
+// Analytics
+
+add_action('init', 'create_tapestry_analytics_schema');
+add_action('init', 'create_tapestry_guid_cookie');
+add_action('wp_ajax_nopriv_tapestry_tool_log_event', 'tapestry_tool_log_event');
+add_action('wp_ajax_tapestry_tool_log_event', 'tapestry_tool_log_event');
+
+/**
+* Create a uuid
+* 
+* @return string
+*/
+function create_tapestry_guid() {
+    $uuid = preg_replace_callback('/x/', function(){
+        $rand = (mt_rand() / mt_getrandmax() * 16 | 0) ^ 0x3;
+        $rand = base_convert ($rand, 10, 16);
+        return strval($rand);
+      }, "xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx");
+    $uuid = preg_replace_callback('/y/', function(){
+        $rand = (mt_rand() / mt_getrandmax() * 16 | 0) ^ 0x8;
+        $rand = base_convert ($rand, 10, 16);
+        return strval($rand);
+      }, $uuid);
+    return $uuid;    
+}
+
+function create_tapestry_guid_cookie() {
+    if (!isset($_COOKIE['tapestry_guid'])) {
+        $uuid = create_tapestry_guid();
+        setcookie('tapestry_guid', $uuid, time()+31556926);
```

Comment:
@jaskwong @wynnset Now that we're storing cookies, will we need to add one of those warning/banners for the user to approve of storing cookies?

Reply:
I don't think we need that. To my best knowledge, those notices are only needed when third-party cookies are utilized in the EU, whereas these are our own cookies used for statistical reasons.